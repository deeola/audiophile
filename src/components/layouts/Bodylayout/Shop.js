import React from "react";
import {Link} from 'react-router-dom'


function Shop(props) {
  return (
    <section className="firstSection">
      <div className="shopSection">
        <div className="shopImageDiv imageheadphones">
          <img className='desktop' src={require('../../../assets/shared/desktop/imageheadphones.png').default} alt='imageheadphones'></img>
        </div>

        <div className='shopTextDiv'>
          <p className='shoptitle'>Headphones</p>
          <p className='shopText'>
            Shop <Link className='shopTextLink' to= '/headphones'><span>&#62;</span> </Link> 
          </p>
        </div>
      </div>
      <div className="shopSection">
        <div className="shopImageDiv imagespeakers">
          <img src={require('../../../assets/shared/desktop/imagespeakers.png').default} alt='imagespeakers'></img>
          
        </div>

        <div className='shopTextDiv'>
          <p className='shoptitle'>Speakers</p>
          <p className='shopText'>
            Shop <Link className='shopTextLink' to='/speakers'> <span>&#62;</span> </Link> 
          </p>
        </div>
      </div>
      <div className="shopSection">
        <div className="shopImageDiv  imageearphones">
          <img src={require('../../../assets/shared/desktop/imageearphones.png').default} alt='imageearphones'></img>

        </div>

        <div className='shopTextDiv'>
          <p className='shoptitle'>Earphones</p>
          <p className='shopText'>
            Shop <Link className='shopTextLink' to='/earphones'> <span>&#62;</span> </Link> 
          </p>
        </div>
      </div>
    </section>
  );
}

export default Shop;
