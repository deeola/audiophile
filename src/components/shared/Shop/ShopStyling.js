import styled from "styled-components";
import { Colors } from "../../../Themes";

export const ShopContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin: 40px auto;
  background: ${Colors.white};

  @media (max-width: 490px) {
    flex-direction: column;
   
  }
`;
