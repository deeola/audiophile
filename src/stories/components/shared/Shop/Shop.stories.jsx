import React from "react";
import { BrowserRouter } from "react-router-dom";
import Shop from "../../../../components/shared/Shop/Shop";
export default {
    title: "Components/Shop",
    component: Shop,
   
  };

  const Template = (args) => (
    <BrowserRouter>
      <Shop {...args} />
    </BrowserRouter>
  );
  
  export const Primary = Template.bind({});
  Primary.args = {
    primary: true,
  };

