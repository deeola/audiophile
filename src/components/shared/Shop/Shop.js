import React from "react";
import imageheadphone from "../../../assets/shared/desktop/imageheadphones.png";
import imagespeaker from "../../../assets/shared/desktop/imagespeakers.png";
import imageearphone from "../../../assets/shared/desktop/imageearphones.png";
import SingleMenu from "../MobileMenu/SingleMenu";
import { ShopContainer } from "./ShopStyling";

const Shop = () => {
  return (
    <ShopContainer data-testid="shop">
      <SingleMenu
        mobile
        label="HEADPHONES"
        img={imageheadphone}
        alt={"headphone"}
        link={"/headphones"}
        datatestid="shopheadphone"
      />
      <SingleMenu
        mobile
        label="SPEAKERS"
        img={imagespeaker}
        alt={"speaker"}
        link={"/speakers"}
        datatestid="shopspeakers"
      />
      <SingleMenu
        mobile
        label="EARPHONES"
        img={imageearphone}
        alt={"earphone"}
        link={"/earphones"}
        datatestid="shopearphones"
      />
    </ShopContainer>
  );
};

export default Shop;
