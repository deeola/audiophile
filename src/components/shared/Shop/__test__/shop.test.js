import Shop from "../Shop";


import React from "react";

import renderer from "react-test-renderer";
import { render, cleanup } from "@testing-library/react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";

it("renders div correctly", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Shop />
    </BrowserRouter>,
    div
  );
});

it("renders 3 items", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Shop data-testid="shop"  />
    </BrowserRouter>
  );

  expect(getByTestId("shop").childElementCount).toBe(3);
});


//snapshot

it('matches snapshot', () => {
    const tree = renderer.create(
        <BrowserRouter>
      <Shop data-testid="shop" primary="primary" />
    </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot()
})