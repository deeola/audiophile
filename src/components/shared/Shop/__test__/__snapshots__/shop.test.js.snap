// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-gKXOVf gxdfrb"
  data-testid="shop"
>
  <div
    className="sc-dkzDqf dIMyJo"
    data-testid="singlemenu"
  >
    <img
      alt="headphone"
      className="sc-eCYdqJ jebgBi"
      data-testid="alt"
      src="imageheadphones.png"
    />
    <p
      className="sc-hKMtZM"
      data-testid="textlabel"
    >
      HEADPHONES
    </p>
    <a
      className="sc-bczRLJ hcrMJw"
      color="tertiary"
      data-testid="shopheadphone"
      hover="tertiary"
      href="/headphones"
      onClick={[Function]}
    >
      Shop
      <span>
        &gt;
      </span>
    </a>
  </div>
  <div
    className="sc-dkzDqf dIMyJo"
    data-testid="singlemenu"
  >
    <img
      alt="speaker"
      className="sc-eCYdqJ jebgBi"
      data-testid="alt"
      src="imagespeakers.png"
    />
    <p
      className="sc-hKMtZM"
      data-testid="textlabel"
    >
      SPEAKERS
    </p>
    <a
      className="sc-bczRLJ hcrMJw"
      color="tertiary"
      data-testid="shopspeakers"
      hover="tertiary"
      href="/speakers"
      onClick={[Function]}
    >
      Shop
      <span>
        &gt;
      </span>
    </a>
  </div>
  <div
    className="sc-dkzDqf dIMyJo"
    data-testid="singlemenu"
  >
    <img
      alt="earphone"
      className="sc-eCYdqJ jebgBi"
      data-testid="alt"
      src="imageearphones.png"
    />
    <p
      className="sc-hKMtZM"
      data-testid="textlabel"
    >
      EARPHONES
    </p>
    <a
      className="sc-bczRLJ hcrMJw"
      color="tertiary"
      data-testid="shopearphones"
      hover="tertiary"
      href="/earphones"
      onClick={[Function]}
    >
      Shop
      <span>
        &gt;
      </span>
    </a>
  </div>
</div>
`;
